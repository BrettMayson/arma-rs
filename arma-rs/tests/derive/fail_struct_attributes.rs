use arma_rs::{FromArma, IntoArma};

#[derive(FromArma, IntoArma)]
#[arma]
struct NoList {
    test: u32,
}

#[derive(FromArma, IntoArma)]
#[arma("literal")]
struct Literal {
    test: u32,
}

#[derive(FromArma, IntoArma)]
#[arma(unknown)]
struct Unknown {
    test: u32,
}

#[derive(FromArma, IntoArma)]
#[arma(unknown::path)]
struct UnknownPath {
    test: u32,
}

#[derive(FromArma, IntoArma)]
#[arma(default, default)]
struct Duplicate {
    test: u32,
}

#[derive(FromArma, IntoArma)]
#[arma(default)]
#[arma(default)]
struct StackedDuplicate {
    test: u32,
}

#[derive(FromArma, IntoArma)]
struct FieldUnknown {
    #[arma(unknown)]
    test: u32,
}

#[derive(FromArma, IntoArma)]
struct FieldDuplicate {
    #[arma(default, default)]
    test: u32,
}

#[derive(FromArma, IntoArma)]
struct FieldStackedDuplicate {
    #[arma(default)]
    #[arma(default)]
    test: u32,
}

#[derive(FromArma, IntoArma)]
#[arma(unknown, default, default)]
struct MultipleErrors {
    test: u32,
}

fn main() {}
