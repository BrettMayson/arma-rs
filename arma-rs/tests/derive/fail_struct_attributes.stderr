error: expected attribute arguments in parentheses: #[arma(...)]
 --> tests/derive/fail_struct_attributes.rs:4:3
  |
4 | #[arma]
  |   ^^^^

error: unexpected literal in nested attribute, expected ident
  --> tests/derive/fail_struct_attributes.rs:10:8
   |
10 | #[arma("literal")]
   |        ^^^^^^^^^

error: unknown arma container attribute `unknown`
  --> tests/derive/fail_struct_attributes.rs:16:8
   |
16 | #[arma(unknown)]
   |        ^^^^^^^

error: unknown arma container attribute `unknown::path`
  --> tests/derive/fail_struct_attributes.rs:22:8
   |
22 | #[arma(unknown::path)]
   |        ^^^^^^^^^^^^^

error: duplicate arma attribute `default`
  --> tests/derive/fail_struct_attributes.rs:28:17
   |
28 | #[arma(default, default)]
   |                 ^^^^^^^

error: duplicate arma attribute `default`
  --> tests/derive/fail_struct_attributes.rs:35:8
   |
35 | #[arma(default)]
   |        ^^^^^^^

error: unknown arma field attribute `unknown`
  --> tests/derive/fail_struct_attributes.rs:42:12
   |
42 |     #[arma(unknown)]
   |            ^^^^^^^

error: duplicate arma attribute `default`
  --> tests/derive/fail_struct_attributes.rs:48:21
   |
48 |     #[arma(default, default)]
   |                     ^^^^^^^

error: duplicate arma attribute `default`
  --> tests/derive/fail_struct_attributes.rs:55:12
   |
55 |     #[arma(default)]
   |            ^^^^^^^

error: unknown arma container attribute `unknown`
  --> tests/derive/fail_struct_attributes.rs:60:8
   |
60 | #[arma(unknown, default, default)]
   |        ^^^^^^^

error: duplicate arma attribute `default`
  --> tests/derive/fail_struct_attributes.rs:60:26
   |
60 | #[arma(unknown, default, default)]
   |                          ^^^^^^^
